homeassistant/components/hassio.py

```diff
@@ -0,0 +1,262 @@
+    for base in ('host', 'homeassistant'):
+        hass.http.register_view(HassIOBaseView(hassio, base))
+    for base in ('supervisor', 'network'):
+        hass.http.register_view(HassIOBaseEditView(hassio, base))
+
+    # register view for addons
+    hass.http.register_view(HassIOAddonsView(hassio))
+
+    @asyncio.coroutine
+    def async_service_handler(service):
+        """Handle HassIO service calls."""
+        addon = service.data.get(ATTR_ADDON)
+        if ATTR_VERSION in service.data:
+            version = {ATTR_VERSION: service.data[ATTR_VERSION]}
+        else:
+            version = None
+
+        # map to api call
+        if service.service == SERVICE_HOST_UPDATE:
+            yield from hassio.send_command(
+                "/host/update", payload=version)
+        elif service.service == SERVICE_HOST_REBOOT:
+            yield from hassio.send_command("/host/reboot")
+        elif service.service == SERVICE_HOST_SHUTDOWN:
+            yield from hassio.send_command("/host/shutdown")
+        elif service.service == SERVICE_SUPERVISOR_UPDATE:
+            yield from hassio.send_command(
+                "/supervisor/update", payload=version)
+        elif service.service == SERVICE_HOMEASSISTANT_UPDATE:
+            yield from hassio.send_command(
+                "/homeassistant/update", payload=version)
+        elif service.service == SERVICE_ADDON_INSTALL:
+            yield from hassio.send_command(
+                "/addons/{}/install".format(addon), payload=version)
+        elif service.service == SERVICE_ADDON_UNINSTALL:
+            yield from hassio.send_command(
+                "/addons/{}/uninstall".format(addon))
+        elif service.service == SERVICE_ADDON_START:
+            yield from hassio.send_command("/addons/{}/start".format(addon))
+        elif service.service == SERVICE_ADDON_STOP:
+            yield from hassio.send_command("/addons/{}/stop".format(addon))
+        elif service.service == SERVICE_ADDON_UPDATE:
+            yield from hassio.send_command(
+                "/addons/{}/update".format(addon), payload=version)
+
+    descriptions = yield from hass.loop.run_in_executor(
+        None, load_yaml_config_file, os.path.join(
+            os.path.dirname(__file__), 'services.yaml'))
+
+    for service, schema in SERVICE_MAP.items():
+        hass.services.async_register(
+            DOMAIN, service, async_service_handler,
+            descriptions[DOMAIN][service], schema=schema)
+
+    return True
+
+
+class HassIO(object):
+    """Small API wrapper for HassIO."""
+
+    def __init__(self, loop, websession):
+        """Initialze HassIO api."""
+        self.loop = loop
+        self.websession = websession
+        try:
+            self._ip = os.environ['HASSIO']
+        except KeyError:
+            self._ip = None
+
+    @property
+    def connected(self):
+        """Return True if it connected to HassIO supervisor."""
+        return self._ip is not None
```

Comment:
Should there be a no-op API command so that we can connect to HASSIO and verify that we can make a connection? We do something similar for our normal API https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/remote.py#L81-L86